Drivers/Src/Lights.c

```diff
@@ -51,42 +81,66 @@ State Lights_Read(light_t light) {
-        portc |= (state << BRAKELIGHT_PIN); // Set value to inputted state
-    } else {
-        uint8_t currData = rxBuf[1];
-        uint8_t newData = currData & ~(0x01 << light);
-        newData |= state << light;
-        txBuf[0] = SPI_OPCODE_W;
-        txBuf[2] = newData;
+    if(lightCurrentState != state){     // Check if state has changed
+        uint16_t lightNewStates = lightStates;
+        
+        lightNewStates &= ~(0x01 << light);
+        lightNewStates |= (state << light);
+        
+        uint8_t txWriteBuf[3] = {SPI_OPCODE_W, SPI_GPIOB, 0x00};
+        uint16_t portc = BSP_GPIO_Read(LIGHTS_PORT);
+
+        if (light == BrakeLight) {
+            portc &= ~(0x01 << BRAKELIGHT_PIN); // Clear bit corresponding to pin
+            portc |= (state << BRAKELIGHT_PIN); // Set value to inputted state
+        } else {
+            txWriteBuf[2] = lightNewStates;
+            
+            // Lights that are external and internal
+            switch (light) {
+                case LEFT_BLINK:
+                    portc &= ~(0x01 << LEFT_BLINK_PIN); // Clear bit corresponding to pin
+                    portc |= (state << LEFT_BLINK_PIN); // Set value to inputted state
+                    break;
+                case RIGHT_BLINK:
+                    portc &= ~(0x01 << RIGHT_BLINK_PIN);    // Clear bit corresponding to pin
+                    portc |= (state << RIGHT_BLINK_PIN);    // Set value to inputted state
+                    break;
+                case Headlight_ON:
+                    portc &= ~(0x01 << HEADLIGHT_PIN);  // Clear bit corresponding to pin
+                    portc |= (state << HEADLIGHT_PIN);  // Set value to inputted state
+                    break;
+                default:
+                    break;
+            }
+        }
+        OSMutexPend(&lightMutex, 0, OS_OPT_PEND_BLOCKING, &timestamp, &err);    // Lock mutex in order to update our lightStates and write to SPI
+        if(err != OS_ERR_NONE){
+            assertOSError(OS_BLINK_LIGHTS_LOC, err);
+        }
+        
+        lightStates = lightNewStates;   // Update lightStates bitmap
 
-        // Lights that are external and internal
-        switch (light) {
-            case LEFT_BLINK:
-                portc &= ~(0x01 << LEFT_BLINK_PIN); // Clear bit corresponding to pin
-                portc |= (state << LEFT_BLINK_PIN); // Set value to inputted state
-                break;
-            case RIGHT_BLINK:
-                portc &= ~(0x01 << RIGHT_BLINK_PIN);    // Clear bit corresponding to pin
-                portc |= (state << RIGHT_BLINK_PIN);    // Set value to inputted state
-                break;
-            case Headlight_ON:
-                portc &= ~(0x01 << HEADLIGHT_PIN);  // Clear bit corresponding to pin
-                portc |= (state << HEADLIGHT_PIN);  // Set value to inputted state
-                break;
-            default:
-                break;
+        GPIO_WriteBit(GPIOA, GPIO_Pin_4, Bit_RESET);
```

Comment:
Is this your chip select pin?

This is probably beyond the scope of this PR, but I think it would be nice if this was more explicit (eg. have a function called `BSP_SPI_SetCS()`)